### PR TITLE
Sentence pair tasks interface

### PR DESCRIPTION
**Pull request by kaustubhdhole**
Add interface and a sample transformation for sentence pair tasks (like paraphrase/entailment)

### CODE REVIEW

File: tasks/TaskTypes.py [hunk 1]
```diff
@@ -20,3 +20,6 @@ class TaskType(enum.Enum):
     SENTIMENT_ANALYSIS = (
         12,
     )  # This is a specific type of text classification with unique properties
+    PARAPHRASE_DETECTION = (13,)
+    TEXTUAL_ENTAILMENT = (14,)
+
```
Comment (Nickeilf) [inline tasks/TaskTypes.py:25]: I think quality estimation can be another task of SentencePairOperation, where the inputs are source and translation and the output is a continuous score.

File: transformations/lexical_counterfactual_generator/transformation.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        return [sentence]
+
+    def generate(self, sentence1: str, sentence2: str, target: str) -> List[Tuple[str, str, str]]:
+        """
+        This implementation augments data for semantic similarity tasks.
+        This converts pairs which are positive (i.e. paraphrases-> label:1) to non-paraphrases (label:0).
+        """
+        examples = []
+        if target == self.pos_label:
+            for sentence1_n in self.negate_if_single_verb(sentence1):
```
Comment (Nickeilf) [inline transformations/lexical_counterfactual_generator/transformation.py:70]: It seems that the `negate_if_single_verb` returns a list of only one sentence, then there's no need to do a for loop here.

File: tasks/TaskTypes.py (hunk 1 above)
Comment (kaustubhdhole) [reply tasks/TaskTypes.py:25]: Excellent idea! I will add it!

File: transformations/lexical_counterfactual_generator/transformation.py (hunk 2 above)
Comment (kaustubhdhole) [reply transformations/lexical_counterfactual_generator/transformation.py:70]: The `bt.generate()` function can generate multiple ones actually.. (line 61)

**Pull request merged**